app/src/main/java/edu/ksu/cs/mdc/calculator/MainActivity.kt

```diff
@@ -2,10 +2,131 @@ package edu.ksu.cs.mdc.calculator
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import kotlin.math.roundToInt
 
 class MainActivity : AppCompatActivity() {
+
+    private var equation : String = ""
+    private var result : String = ""
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
+
+    /**
+     * Called by buttons for numbers and operators.
+     *
+     * Fills the equation variable and displays it.
+     */
+    fun buttonPress(view: android.view.View) {
+        val btn : Button? = view as? Button
+        val btntext : String? = btn?.text.toString()
```

Comment:
btntext is always non-null, so you don't need to have the question mark on "String?"